evm/src/executor/inspector/cheatcodes/util.rs

```diff
@@ -324,6 +327,20 @@ pub fn parse_private_key(private_key: U256) -> Result<SigningKey, Bytes> {
     SigningKey::from_bytes(&bytes).map_err(|err| err.to_string().encode().into())
 }
 
+// Determines if the gas limit on a given call was manually set in thes cript and should therefore
+// not be overwritten by later estimations
+pub fn check_if_fixed_gas_limit<DB: DatabaseExt>(
+    data: &EVMData<'_, DB>,
+    call_gas_limit: u64,
+) -> bool {
+    // If the gas limit was not set in the source code it is set to the estimated gas left at the
+    // time of the call, which should be rather close to configured gas limit.
+    // TODO: Find a way to reliably make this determenation. (for example by
+    // generating it in the compilation or evm simulation process)
+    U256::from(data.env.tx.gas_limit) > data.env.block.gas_limit &&
```

Comment:
I don't think that would work due to the following reason:

The call_gas_limit (which comes from the CallInput struct returned by revm is always set to some value. In the case of "estimated" gas it will be set to all the remaining gas at the time of the call as determined by revm. From my understanding this is calculated based on the configured gas_imit (which is set to i32.max by default and does not take the senders eth balance into account).

So if we just always pass through the gas_limit we will end up with values that are way too high and corresponding transaction failures when submitting the transaction.

Basically this is the issue I tried to describe (way too verbosely) in my [comment here](https://github.com/foundry-rs/foundry/pull/4219#issuecomment-1407601873).

Reply:
In short the answer is: "Yes we need this function, but we should probably find a better way to determine wether the gas_limit was set manually."